module/zfs/zfs_znode.c

```diff
@@ -113,9 +113,8 @@ zfs_znode_cache_constructor(void *buf, void *arg, int kmflags)
 	mutex_init(&zp->z_acl_lock, NULL, MUTEX_DEFAULT, NULL);
 	rw_init(&zp->z_xattr_lock, NULL, RW_DEFAULT, NULL);
 
-	mutex_init(&zp->z_range_lock, NULL, MUTEX_DEFAULT, NULL);
-	avl_create(&zp->z_range_avl, zfs_range_compare,
-	    sizeof (rl_t), offsetof(rl_t, r_node));
+	zfs_rlock_init(&zp->z_range_lock, &zp->z_size, &zp->z_blksz,
+	    &ZTOZSB(zp)->z_max_blksz);
```

Comment:
The assignment could be done in `zfs_inode_alloc()` the caller of the constructor where `sb->sb->s_fs_info` is available.